### PR TITLE
[Hub Generated] Review request for Microsoft.WindowsESU to add version preview/2019-09-16-preview

### PR DESCRIPTION
**Pull request by mbialecka**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/mbialecka/azure-rest-api-specs/dev-windowsesu-Microsoft.WindowsESU-2019-09-16-preview...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by mbialecka** [comment]
The spec was already approved in the private repo:
https://github.com/Azure/azure-rest-api-specs-pr/pull/948
https://github.com/Azure/azure-rest-api-specs-pr/pull/956

**Comment by fengzhou-msft** [comment]
@NullMDR can you take a look at the Go SDK automation failure?

**Comment by NullMDR** [comment]
@ArcturusZhang Could you please take a look at the go generation failure? Seems like the autorest.go has some problem.

File: specification/windowsesu/resource-manager/readme.go.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  clear-output-folder: true
+```
+
+### Tag: package-2019-09-16-preview and go
+
+These settings apply only when `--tag=package-2019-09-16-preview --go` is specified on the command line.
+Please also specify `--go-sdks-folder=<path to the root directory of your azure-sdk-for-go clone>`.
+
+```yaml $(tag) == 'package-2019-09-16-preview' && $(go)
+namespace: Microsoft.WindowsESU
```
Comment (ArcturusZhang) [inline specification/windowsesu/resource-manager/readme.go.md]: please change the namespace to `windowsesu`

File: specification/windowsesu/resource-manager/readme.go.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+output-folder: $(go-sdks-folder)/windowsesu/Generated
```
Comment (ArcturusZhang) [inline specification/windowsesu/resource-manager/readme.go.md]: please change the `output-folder` to
`$(go-sdk-folder)/services/preview/windowsesu/2019-09-16-preview/$(namespace)`

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]

File: specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+  },
+  "definitions": {
+    "MultipleActivationKey": {
+      "description": "MAK key details.",
+      "type": "object",
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
```
Comment (KrisBash) [inline specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json]: It's best to carve the common resource properties into a shared definition. Better yet, use an allOf reference to link the common definition from common-types. https://github.com/Azure/azure-rest-api-specs/blob/a6f41b54a296e11631a65bbe4a2da623afb758a2/specification/common-types/resource-management/v1/types.json#L45

File: specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+              "default": "SupplementalServicing",
+              "x-ms-enum": {
+                "name": "SupportType",
+                "modelAsString": true
+              },
+              "x-ms-mutability": ["create", "read"]
+            },
+            "installedServerNumber": {
+              "description": "Number of activations/servers using the MAK key.",
+              "type": "integer",
```
Comment (KrisBash) [inline specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json]: This is user set or read-only?

**Review by KrisBash — APPROVED** [review]
Signing off with a few comments

**Review by ArcturusZhang — APPROVED** [review]
LGTM for the go SDK part

File: specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json (hunk 4 above)
Comment (mbialecka) [reply specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json]: Set by user on creation, immutable, as most of other properties.

File: specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json (hunk 3 above)
Comment (mbialecka) [reply specification/windowsesu/resource-manager/Microsoft.WindowsESU/preview/2019-09-16-preview/windowsesu.json]: I've added references to common types

**Pull request merged**